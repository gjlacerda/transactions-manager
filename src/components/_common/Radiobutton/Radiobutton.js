import React from 'react'
import styled from 'styled-components'
import { colorBlack, colorCredit, colorDebit } from 'utils/color'
import { opacityLight } from 'utils/opacity'
import { transitionInput } from 'utils/transition'
import { fontSize } from 'utils/font'

export const getColor = (props) => {
  if (props.credit) {
    return colorCredit
  }
  if (props.debit) {
    return colorDebit
  }
  return colorBlack
}

const radioWidth = '20px'

const Container = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;

  &:hover label {
    opacity: 1;
  }

  &.active {
    .fakeinput {
      border-color: ${props => getColor(props)};
    }

    .dot {
      background-color: ${props => getColor(props)};
    }
  }
`

const Fakeinput = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${radioWidth};
  height: ${radioWidth};
  border: 2px solid ${colorBlack};
  border-radius: 50%;
`

const Label = styled.label`
  font-size: ${fontSize.small};
  margin-left: 8px;
  opacity: ${opacityLight};
  cursor: pointer;
  transition ${transitionInput('opacity')};
`

const dotWidth = '10px'

const Dot = styled.div`
  width: ${dotWidth};
  height: ${dotWidth};
  border-radius: 50%;
`

export default ({
  label,
  credit,
  debit,
  active,
  onClick,
}) => (
  <Container
    className={active ? 'active' : ''}
    credit={credit}
    debit={debit}
    onClick={onClick}
  >
    <Fakeinput className="fakeinput">
      <Dot className="dot" />
    </Fakeinput>
    <Label>{label}</Label>
  </Container>
)
