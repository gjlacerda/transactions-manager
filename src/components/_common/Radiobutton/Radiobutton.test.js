import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Radiobutton from './Radiobutton'

describe('Radiobutton component', () => {
  it('deve ser igual ao snapshot', () => {
    const tree = renderer.create(
      <Radiobutton
        label="Teste"
        credit
        active
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('deve chamar a função onClick passada por parâmetro', () => {
    const spy = jest.fn()
    const wrapper = shallow(<Radiobutton onClick={spy} />)
    wrapper.simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})
