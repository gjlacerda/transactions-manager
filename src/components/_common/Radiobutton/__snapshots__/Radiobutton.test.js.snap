// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radiobutton component deve ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:hover label {
  opacity: 1;
}

.c0.active .fakeinput {
  border-color: #56caa4;
}

.c0.active .dot {
  background-color: #56caa4;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  border: 2px solid #555;
  border-radius: 50%;
}

.c3 {
  font-size: 14px;
  margin-left: 8px;
  opacity: 0.8;
  cursor: pointer;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
}

.c2 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
}

<div
  className="active c0"
>
  <div
    className="fakeinput c1"
  >
    <div
      className="dot c2"
    />
  </div>
  <label
    className="c3"
  >
    Teste
  </label>
</div>
`;
